Add support for * in workspace symbol search

gayanper opened the pull request:
The new changset fix the issue with * in workspace symbol search by passing the original query with * in the LS, but use query components without * for matching with fuzzy score.

Improvement to resolve : https://github.com/microsoft/vscode/issues/149144

ProKashif commented on src/vs/base/common/fuzzyScorer.ts:
```diff
@@ -871,18 +876,19 @@ export function prepareQuery(original: string): IPreparedQuery {
 
 				values.push({
-					original: originalPiece,
+					original: originalPiece.replace('*', '').replace('?', ''),
```
```suggestion
					original: originalPiece?.replace('*', '')?.replace('?', ''),
```

gayanper replied:
Thanks for taking a look at the code. But I don't see a need for using optional chaining here since originalPiece variable cannot be null nor undefined right ? I making the assumption based on the rest of the statements which use the same variable. Do you see otherwise ?

ProKashif commented on src/vs/base/common/fuzzyScorer.ts:
```diff
@@ -871,18 +876,19 @@ export function prepareQuery(original: string): IPreparedQuery {
 	}
-
-	return { original, originalLowercase, pathNormalized, normalized, normalizedLowercase, values, containsPathSeparator, expectContiguousMatch: expectExactMatch };
+	original = original.replace('*', '').replace('?', '');
```
```suggestion
	original = original?.replace('*', '')?.replace('?', '');
```

gayanper replied:
Thanks for taking a look at the code. But I don't see a need for using optional chaining here since original variable cannot be null nor undefined right ? I making the assumption based on the rest of the statements which use the same variable. Specially looking at line 851. Do you see otherwise ?

gayanper commented:
@TylerLeonhardt will you be able to have a look at this change and let me know what you think ?

gayanper commented:
@jrieken will you have time to check this change?

jrieken commented:
No, I am the wrong person for this

gayanper commented:
> No, I am the wrong person for this

@jrieken could help me to reach the correct leads in this area to get an review ?

gayanper commented:
I think we can improve this change not to be intrusive on fuzzy score login, will see if the problem could be solved without replacing * and ? and try to solve it at the symbol picker side instead.

gayanper committed:
Add support for * in workspace symbol search

The fix change the code so that the query to LS is sent with * while the
query components used to perform the matching is fixed by removing the
* characters for workspace symbol search.

gayanper commented:
@TylerLeonhardt I have update the PR with a new approach to solve the * issue. Please let me know what you think.

gayanper commented:
Any feedback on this PR ?

TylerLeonhardt commented:
Hey 👋 I haven't had the time to look at this PR. I'll try to get to it in November (I'll be going out of town for 2 weeks)

gayanper commented:
Any update on this ?

raulvaldoleiros commented:
-- please?

gayanper commented:
@TylerLeonhardt any updates on this ? Will this be considered for next release in 2023 ?

gayanper commented:
Any update on this ?

vivodi commented:
Is this still relevant?

raulvaldoleiros commented:
Yes, this continues to be relevant because without it makes search for symbols quite hard
